src/GNDStk/gnds-keyword.hpp

```diff
@@ -127,53 +144,187 @@ namespace detail {
 // keyword_t
 class keyword_t {
 public:
+
    // ------------------------
    // meta
    // ------------------------
 
-   // <[T]>
-   template<class T = void>
+   // <>
+   // <RESULT>
+   template<class RESULT = void>
    static auto meta(const std::string &name)
    {
-      return meta_t<T>{name};
+      return meta_t<RESULT>{name};
    }
 
    // ------------------------
    // child
    // ------------------------
 
-   // <[T]>
-   template<class T = void>
-   static auto child(const std::string &name)
+   /*
+   // meta examples
+   inline const auto encoding = keyword.meta<>("encoding");
+   inline const auto encoding = keyword.meta<encoding_t>("encoding");
+
+   // child
+   inline const auto axes = keyword.child<            >("axes");
+   inline const auto axes = keyword.child<axes_t      >("axes");
+   inline const auto axes = keyword.child<axes_t,false>("axes");
+
+   inline const auto axis = keyword.child<void,  true >("axis");
+   inline const auto axis = keyword.child<axis_t,true >("axis");
+   */
+
+   // Forms:
+   //
+   //    child<(void),(false) >(name    )   Node<>, singular, name, no
+   //    child<(void),(false) >(name,top)   Node<>, singular, name, top-level?
+   //    child<RESULT         >(name    )   RESULT, singular, name, no
+   //    child<RESULT         >(name,top)   RESULT, singular, name, top-level?
+   //    child<RESULT,MULTIPLE>(name    )   RESULT, multiple, name, no
+   //    child<RESULT,MULTIPLE>(name,top)   RESULT, multiple, name, top-level?
+   //
+   // Allowable metadata and children are unspecified; therefore we use:
+   //
+   //    metadata<>
+   //    children<>
+   //
+   // meaning that any are allowed.
+   //
+   template<class RESULT = void, bool MULTIPLE = false>
+   static auto child(const std::string &name, const bool top = false)
+   {
+      return child_t<
+         RESULT,
+         MULTIPLE,
+         metadata<>, // any
+         children<>  // any
+      >{name,top};
+   }
+
+   // Upcoming variations:
+   //
+   //    template<class RESULT,                class K, class... Ks>
+   //    template<class RESULT,                class K, class... Ks>
+   //    template<class RESULT, bool MULTIPLE, class K, class... Ks>
+   //    template<class RESULT, bool MULTIPLE, class K, class... Ks>
+   //
+   //    static auto child(const string &name,                 const K &, Ks...)
+   //    static auto child(const string &name, const bool top, const K &, Ks...)
+   //    static auto child(const string &name,                 const K &, Ks...)
+   //    static auto child(const string &name, const bool top, const K &, Ks...)
+   //
+   // As-written above, in order to see the structure.
+
+   // Form:
+   //    child<RESULT,(false) >(name,keywords)
+   template<class RESULT, class K, class... Ks>
+   static auto child(const std::string &name, const K &, Ks...)
```

Comment:
I'm really not sure what all the different templates are for here.

Reply:
When I get time to finish these ideas, it'll be much more clear why this stuff is here at all.